src/Bridges/ApplicationLatte/TemplateFactory.php

```diff
@@ -112,4 +115,12 @@ public function createTemplate(UI\Control $control = NULL)
 		return $template;
 	}
 
+	public function setTemplateClass($templateClass)
+	{
+		if (!class_exists($templateClass) || !is_a($templateClass, UI\ITemplate::class, TRUE)) {
```

Comment:
My guess is that it's because `TemplateFactory` has `@return Template` annotation. However Nette itself does not seem to rely on the instance being `Template`.